js/index.js

```diff
@@ -37,6 +37,73 @@ const siteContent = {
   },
 };
 
-// Example: Update the img src for the logo
-let logo = document.getElementById("logo-img");
-logo.setAttribute('src', siteContent["nav"]["img-src"])
+// helper functions 
+
+function ObjToArray(array){
+  return Object.entries(array); // returns array of key pair values
+}
+
+function isContentHeader(value){
+  return value[0].includes("h4");
+}
+
+function notContentHeader(value) {
+  return !value[0].includes("h4")
+}
+
+function populateHeaderMenu(menu, image) {
+
+}
+
+function populateMainContent(oldContent, newHeaders, newText) {
+  let headerContent = oldContent.querySelectorAll(".text-content h4")
+  for (let i = 0; i <= headerContent.length; i++) {
+    headerContent[i].textContent = newHeaders[i];
+  }
+
+  let textContent = oldContent.querySelectorAll(".text-content p")
+  for (let i = 0; i <= headerContent.length; i++) {
+    headerContent[i].textContent = newText[i];
+  }
+}
+
+// Step 1 // declare data & relevant properties 
+
+let navItems = siteContent.nav;
+
+let header = document.querySelector("header");
+let nav = document.querySelector("header nav");
+let listOfLinks = nav.querySelectorAll("a");
+
+let ctaItems = siteContent.cta;
+let cta = document.querySelector(".cta");
+
+let main_content_items = siteContent["main-content"]; 
+let maincontent = document.querySelector(".main-content");
+console.log(maincontent);
+
+let contactItems = siteContent.contact;
+let contact = document.querySelector(".contact");
+
+let footerItem = siteContent.footer;
+let footer = document.querySelector(".footer");
+
+// Step 2 // filter out by property / class.
+
+const navImg = ObjToArray(navItems).filter(e => e[0].includes("img-src"));
```

Comment:
Here again the `Object.keys()` method is so nice since it gives an array of the keys` /// something to think about...